lib/yt/models/content_owner.rb

```diff
@@ -46,6 +46,22 @@ def initialize(options = {})
         @display_name = options[:display_name]
       end
 
+      # Uploads a reference file to YouTube.
+      # @param [String] path_or_url is the video or audio file to upload. Can either be the
+      #   path of a local file or the URL of a remote file.
+      # @param [Hash] params the metadata to add to the uploaded reference.
+      # @option params [String] :asset_id The id of the asset the uploaded reference belongs to.
+      # @option params [String] :content_type The type of content being uploaded.
+      # @return [Yt::Models::Reference] the newly uploaded reference.
+      def upload_reference_file(path_or_url, params = {})
+        file = open path_or_url, 'rb'
```

Comment:
I think this will behave pretty differently if the file comes from a URL vs a path. If it's from a URL, `open` will download the file to a tempfile, whereas if it's a path, we just make a pointer to that file descriptor (and Ruby can ask the OS for its size, etc). Do we care about that?

Reply:
So the main difference is that with a URL, we have a tempfile left over in the directory? It doesn't matter for my use case, as tempfiles get cleaned up automatically in production. This is the same exact code taken from the [upload_video](https://github.com/Fullscreen/yt/blob/master/lib/yt/models/account.rb#L71) method in `account.rb`.